content/mods/E-rust-for-systems/topics/ffi/exercises/qoi-bindgen/description.md

```diff
@@ -0,0 +1,177 @@
+In this exercise, we will use `cargo bindgen` to generate the FFI bindings for a C library. Bindgen will look at a C header file, and generate Rust functions, types and constants based on the C definitions.
+
+However, the generated code will likely be ugly and non-idiomatic. To wrap a C library properly, good API design and documentation is needed. 
+
+### Background
+The [image crate](https://crates.io/crates/image) provides functionality for encoding, decoding and editing images in Rust. It supports many image formats, like JPEG, PNG and GIF, but also QOI. QOI is a "Quite OK Image format", which aims for fast encoding and decoding of images, while providing a file size similar to PNGs.
+In this exercise, we test if the image crate produces the same results when decoding QOI images as the [QOI reference C library](https://github.com/phoboslab/qoi). 
+
+The QOI C library is a header-only library, which means the function implementations are included within the header file instead of in a separate C file. We've added a separate C file which includes the header to make it easier to compile and include the library in our Rust program.
+
+### Generating bindings
```

Comment:
Please use the `#[modmod:exercise_ref]` to enable modmod to number the exercises upon generating. [Example](https://github.com/trifectatechfoundation/teach-rs/blob/ff24c7e3c5a45c3d6ae187abb4baa9843254bbce/content/mods/B-crate-engineering/topics/testing/exercises/fizzbuzz/description.md?plain=1#L7)

Reply:
Would it be possible to hack modmod so it just 'sees' that or any markdown subheading? Or is there use case why we would want to have unnumbered subheadings?